scripts/pylib/twister/twisterlib/reports.py

```diff
@@ -51,34 +53,34 @@ def xunit_testcase(eleTestsuite, name, classname, status, ts_status, reason, dur
             name=f"{name}",
             time=f"{duration}")
 
-        if status in ['skipped', 'filtered']:
+        if status in [TestCaseStatus.SKIP, TestCaseStatus.FILTER]:
             skips += 1
             # temporarily add build_only_as_skip to restore existing CI report behaviour
-            if ts_status == "passed" and not runnable:
+            if ts_status == TestSuiteStatus.PASS and not runnable:
                 tc_type = "build"
             else:
                 tc_type = status
-            ET.SubElement(eleTestcase, 'skipped', type=f"{tc_type}", message=f"{reason}")
-        elif status in ["failed", "blocked"]:
+            ET.SubElement(eleTestcase, ReportStatus.SKIP, type=f"{tc_type}", message=f"{reason}")
```

Comment:
I would leave those sepeare from the twister status, we need to be able to translate in the report from twister status to junit/xunit but I would not manage them at the same level as the other statuses

Reply:
I have excised that class from the rest of the statuses and moved it to `reports.py`. Thus we can achieve clear distinction from the rest of the Status classes while helping code readers know which values are possible.